Client/Client.cpp

```diff
@@ -1,44 +1,109 @@
 #include <iostream>
 #include "Client.hpp"
 
-Client::Client() {}
-
-Client::Client(int port, std::string &name, std::string &nick,
-        std::string &pass) : m_port(port), m_name(name), m_nick(nick), m_pass(pass) {
-
-    m_msg = "";
+Client::Client(const char *port, const char *servaddr)
+    : m_port(port), m_servaddr(servaddr), m_clientsock(-1), m_login(), m_pass() {}
+
+bool Client::Start() {
+    struct addrinfo clientaddr, *result;
+
+    memset(&clientaddr, 0, sizeof(clientaddr));
+    clientaddr.ai_family = AF_UNSPEC;
+    clientaddr.ai_socktype = SOCK_STREAM;
+
+    if (getaddrinfo(m_servaddr, m_port, &clientaddr, &result) < 0) {
+        std::cerr << "getaddrinfo error" << std::endl;
+        return false;
+    }
+
+    m_clientsock = socket(result->ai_family, result->ai_socktype, result->ai_protocol);
+    if (m_clientsock < 0) {
+        std::cerr << "Failed to create the socket" << std::endl;
+        freeaddrinfo(result);
+        return false;
+    }
+
+    if (connect(m_clientsock, result->ai_addr, result->ai_addrlen) < 0) {
+        std::cerr << "Failed to connect to the server" << std::endl;
+        close(m_clientsock);
+        freeaddrinfo(result);
+        return false;
+    }
+
+    freeaddrinfo(result);
+    return true;
 }
 
-void Client::Run() {}
-
-void Client::SignUp() {}
-
-void Client::SignIn() {}
-
-int Client::getPort() const { return m_port; }
-
-std::string Client::getName() const { return m_name; }
-
-std::string Client::getNick() const { return m_nick; }
-
-std::string Client::getPass() const { return m_pass; }
-
-std::string Client::getMsg() const { return m_msg; }
-
-void Client::setName(std::string &name) { m_name = name; }
-
-void Client::setNick(std::string &nick) { m_nick = nick; }
-
-void Client::setPass(std::string &pass) { m_pass = pass; }
+bool Client::SendMsgToServer(const std::string &msg) {
+    if (send(m_clientsock, msg.c_str(), msg.size(), 0) < 0) {
+        std::cerr << "Failed to send a message" << std::endl;
+        return false;
+    }
+    return true;
+}
 
-void Client::setMsg(std::string &msg) { m_msg = msg; }
+bool Client::ReceiveMsgFromServer(std::string &receivedMsg) {
```

Comment:
add recv timeout for example i using polling but you cat configure youre socket and add read timeouts
```
template <typename T>
bool read_data(T data, size_t size, uint16_t timeout)
{
    INFO("reading data, timeout %d.\n", timeout);

    // init poll struct
    struct pollfd mypoll = { device_socket, POLLIN|POLLPRI };

    if (poll(&mypoll, 1, timeout * 1000))
    {
        if(read(device_socket, data, size) < 0)
        {
            ERROR("read error : %d\n", errno);
            return false;
        }

        print_hex_data(data, size, "read_data <<<");
    }
    else
    {
        WARNING("error: read timeout\n");
        return false;
    }

    return true;
}
```
[poll doc](https://man7.org/linux/man-pages/man2/poll.2.html)
[socket recv timeout](https://stackoverflow.com/questions/2876024/linux-is-there-a-read-or-recv-from-socket-with-timeout)